File: plugins/on_audio_query-2.6.1/ios/Classes/utils/OnDeviceInfo.swift
import Flutter

// TODO: Add more specific return to [Model].
// This method will get some basic information about platform.
public func queryDeviceInfo(result: FlutterResult) {
    let device = UIDevice.current
    let deviceData: [String: Any] = [
        "device_model": device.model,
        "device_sys_type": device.systemName,
        "device_sys_version": device.systemVersion,
    ]
    result(deviceData)
}

File: plugins/on_audio_query-2.6.1/android/src/main/kotlin/com/lucasjosino/on_audio_query/utils/OnDeviceInfo.kt
package com.lucasjosino.on_audio_query.utils

import android.os.Build
import io.flutter.plugin.common.MethodChannel

fun queryDeviceInfo(result: MethodChannel.Result) {
    val deviceData: MutableMap<String, Any> = HashMap()
    deviceData["device_model"] = Build.MODEL
    deviceData["device_sys_version"] = Build.VERSION.SDK_INT
    deviceData["device_sys_type"] = "Android"
    result.success(deviceData)
}